packages/jsii/lib/jsii-diagnostic.ts

```diff
@@ -238,6 +238,20 @@ export class JsiiDiagnostic implements ts.Diagnostic {
     name: 'metadata/missing-peer-dependency',
   });
 
+  public static readonly JSII_0006_MISSING_DEV_DEPENDENCY = Code.warning({
```

Comment:
Maybe drop a comment here stating that _for now_, it is not possible to increase the severity of this because it's generated _before_ the severity settings are applied?